executor/union_scan_test.go

```diff
@@ -332,3 +332,28 @@ func (s *testSuite7) TestForUpdateUntouchedIndex(c *C) {
 	tk.MustExec("commit")
 	tk.MustExec("admin check table t")
 }
+
+func (s *testSuite7) TestUpdateScanningHandles(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("create table t(a int primary key, b int);")
+	tk.MustExec("begin")
+	for i := 2; i < 100000; i++ {
+		tk.MustExec("insert into t values (?, ?)", i, i)
+	}
+	tk.MustExec("commit;")
+
+	tk.MustExec("set tidb_distsql_scan_concurrency = 1;")
```

Comment:
Why set those concurrency variables?

Reply:
To slow down the execution, otherwise, the concurrent scanner cannot scan the updated rows.

This SQL is a special case, tries to update `handle + 1000` when reading `headle`. If the executor is slow enough, it can read some dirty data.